src/SAL/nfs4_state_id.c

```diff
@@ -511,7 +511,8 @@ void dec_nfs4_state_ref(struct state_t *state)
 
 	PTHREAD_MUTEX_destroy(&state->state_mutex);
 
-	op_ctx->fsal_export->exp_ops.free_state(op_ctx->fsal_export, state);
+	if ((op_ctx != NULL) && (op_ctx->fsal_export != NULL))
+		op_ctx->fsal_export->exp_ops.free_state(op_ctx->fsal_export, state);
```

Comment:
Should we do nullcheck for exp_ops as well? @derekbit

Reply:
Sounds good and the check is fair. Will patch it.

Reply:
I just merged the share manager PR, so basically you just rebuild the same image so we don't need to have another PR for share manager.

Reply:
Just found the check of `exp_ops` is not necessary, because  `exp_ops` is not a pointer. ([code](https://github.com/longhorn/nfs-ganesha/blob/longhorn-ganesha-v4/src/include/fsal_api.h#L2961))

Reply:
ah, thanks for the update. then all good here.